src/background/tab-state.js

```diff
@@ -48,6 +48,12 @@ export default function TabState(initialState, onchange) {
   const self = this;
   let currentState;
 
+  // This variable contains a map between the tabId and a object with a
+  // cancellation function and the current waiting time.
+  // The cancellation function is used to abort the badge request in a debounce fashion
+  /** @type {Map<number, {cancel:function, waitMs: number}>} */
```

Comment:
It could help here to introduce a `@typedef` which will allow you to document the `cancel` and `waitMs` properties more directly with `@prop`. The variable type can then be `/** @type {Map<number,BadgeRequest>} */` (where `BadgeRequest` is whatever you decide to call the type.